src/main/java/org/mastodon/mamut/feature/spot/ellipsoid/SpotEllipsoidFeature.java

```diff
@@ -0,0 +1,201 @@
+@Plugin( type = FeatureSerializer.class )
+public class SpotEllipsoidFeature implements Feature< Spot >, FeatureSerializer< SpotEllipsoidFeature, Spot >
+{
+
+	public static final String KEY = "Spot ellipsoid parameters";
+
+	private static final String HELP_STRING =
+			"Computes spot ellipsoid parameters, i.e. the 3 semi axes and the volume. ";
+
+	private final Map< FeatureProjectionKey, FeatureProjection< Spot > > projectionMap;
+
+	public static final FeatureProjectionSpec AXIS_A_PROJECTION_SPEC =
+			new FeatureProjectionSpec( "Semi-axis a", Dimension.LENGTH );
+
+	public static final FeatureProjectionSpec AXIS_B_PROJECTION_SPEC =
+			new FeatureProjectionSpec( "Semi-axis b", Dimension.LENGTH );
+
+	public static final FeatureProjectionSpec AXIS_C_PROJECTION_SPEC =
+			new FeatureProjectionSpec( "Semi-axis c", Dimension.LENGTH );
+
+	public static final FeatureProjectionSpec VOLUME_PROJECTION_SPEC =
+			new FeatureProjectionSpec( "Volume", Dimension.NONE );
+
+	final DoublePropertyMap< Spot > semiAxisA;
+
+	final DoublePropertyMap< Spot > semiAxisB;
+
+	final DoublePropertyMap< Spot > semiAxisC;
+
+	final DoublePropertyMap< Spot > volume;
+
+	@Plugin( type = FeatureSpec.class )
+	public static class SpotEllipsoidFeatureSpec extends FeatureSpec< SpotEllipsoidFeature, Spot >
+	{
+		public SpotEllipsoidFeatureSpec()
+		{
+			super( KEY, HELP_STRING, SpotEllipsoidFeature.class, Spot.class, Multiplicity.SINGLE,
+					AXIS_A_PROJECTION_SPEC, AXIS_B_PROJECTION_SPEC, AXIS_C_PROJECTION_SPEC, VOLUME_PROJECTION_SPEC );
+		}
+	}
+
+	public static final SpotEllipsoidFeatureSpec SPOT_ELLIPSOID_FEATURE_SPEC = new SpotEllipsoidFeatureSpec();
+
+	SpotEllipsoidFeature(
+			final DoublePropertyMap< Spot > semiAxisA, final DoublePropertyMap< Spot > semiAxisB,
+			final DoublePropertyMap< Spot > semiAxisC, final DoublePropertyMap< Spot > volume )
+	{
+		this.semiAxisA = semiAxisA;
+		this.semiAxisB = semiAxisB;
+		this.semiAxisC = semiAxisC;
+		this.volume = volume;
+		this.projectionMap = new LinkedHashMap<>( 4 );
+		{
+			final FeatureProjectionKey featureProjectionKeyAxisA = key( AXIS_A_PROJECTION_SPEC );
+			final FeatureProjectionKey featureProjectionKeyAxisB = key( AXIS_B_PROJECTION_SPEC );
+			final FeatureProjectionKey featureProjectionKeyAxisC = key( AXIS_C_PROJECTION_SPEC );
+			final FeatureProjectionKey featureProjectionKeyVolume = key( VOLUME_PROJECTION_SPEC );
+
+			projectionMap.put( featureProjectionKeyAxisA,
+					FeatureProjections.project( featureProjectionKeyAxisA, semiAxisA, Dimension.NONE_UNITS ) );
+			projectionMap.put( featureProjectionKeyAxisB,
+					FeatureProjections.project( featureProjectionKeyAxisB, semiAxisB, Dimension.NONE_UNITS ) );
+			projectionMap.put( featureProjectionKeyAxisC,
+					FeatureProjections.project( featureProjectionKeyAxisC, semiAxisC, Dimension.NONE_UNITS ) );
+			projectionMap.put( featureProjectionKeyVolume,
+					FeatureProjections.project( featureProjectionKeyVolume, volume, Dimension.NONE_UNITS ) );
+		}
```

Comment:
I would prefer the lokal variable names to be even shorter. For example keyShort, keyMiddle, keyLong, keyVolume. This would make the long lines shorter. And the variable names would still contain all the information required to easily understand this eight line of code.

That said. The code is, good enough!